tests/test_utils.cpp

```diff
@@ -52,6 +54,14 @@ void printErrors(const libcellml::Parser &p)
     }
 }
 
+void expectEqualErrors(const std::vector<std::string> &errors, const libcellml::Logger &logger)
+{
+    EXPECT_EQ(errors.size(), logger.errorCount());
+    for (size_t i = 0; i < logger.errorCount() && i < errors.size(); ++i) {
+        EXPECT_EQ(errors.at(i), logger.error(i)->description());
+    }
+}
```

Comment:
This is a useful test utility, but we should then make use of it everywhere we check for expected errors. For example, `tests/validator/validator.cpp` does something similar (e.g. `invalidCellMLIdentifiersWithSpecificationHeading` although this particular test also checks for the specification heading, something that we might want to add).

Reply:
It is a useful utility although there is a downside.  When an error is reported by gtest it gives the line location of the utility file and not from the line which is inside the test.  Small downside I think though.

Reply:
Hmm, I hadn't thought of that, but as long as we can trace back the failing test, then I am ok with that small downside.

Reply:
Yes we have the name of the test which is failing, you just can't use the line number to make the jump.

We could add a default parameter to this function to also check other pieces of information.

Reply:
> Yes we have the name of the test which is failing, you just can't use the line number to make the jump.

Ok, that's better than nothing although if that utility is used many times in a given test then it might be tricky to determine what is wrong. Hmm, although handy, I am wondering whether it's such a good idea to rely on that utility. What about a macro? I don't like the idea of macros, but... it could work here...?

> We could add a default parameter to this function to also check other pieces of information.

If we end up using that utility, in some way or another, then yes that would be good to have.

Reply:
We could use a macro and it would make sense but I think I would have to kill myself.

Reply:
Not that I have strong feelings about macros or anything.

Reply:
Ok, I leave it to you to convert it to a macro... and try not to kill yourself as a result of it. We still need you for a few things...